spatialmath/box.go

```diff
@@ -352,3 +355,93 @@ func separatingAxisTest(positionDelta, plane r3.Vector, halfSizeA, halfSizeB [3]
 	}
 	return sum
 }
+
+// ToPointCloud converts a box geometry into a []r3.Vector. This method takes one argument which
+// determines how many points to place per square mm. If the argument is set to 0. we
+// automatically substitute the value with defaultBoxPointDensity.
+func (b *box) ToPointCloud(resolution float64) []r3.Vector {
+	// check for user defined spacing
+	var iter float64
+	if resolution != 0. {
+		iter = resolution
+	} else {
+		iter = defaultBoxPointDensity
+	}
+
+	// TODO: the fillFaces method calls can be made concurrent if the ToPointCloud method is too slow
+	// the boolean values which are passed into the fillFaces method allow for common edges to be
+	// ignored. This removes duplicate points on the box's surface
+	var facePoints [][]r3.Vector
+	facePoints = append(facePoints, fillFaces(b.halfSize, iter, 0, true, false))
```

Comment:
`facePoints` should be a `[]r3.Vector`, and these appends should use elipses `...` to expand the arguments for append.

Reply:
Example usage:
https://stackoverflow.com/questions/40657553/all-uses-of-ellipsis

Here's some more background on variadic functions such as append:
https://programming.guide/go/three-dots-ellipsis.html